src/main/java/com/qubit/solution/fenixedu/integration/cgd/services/form43/CgdForm43Sender.java

```diff
@@ -169,35 +185,60 @@ public boolean sendForm43For(Registration registration, boolean requestCard) {
-                                        + uploadFormAttachment.getCodeId() + "\n Unique Error ID: "
-                                        + uploadFormAttachment.getUEC()
+                                        + studentNumber + " with message: " + attachmentValueMessage + "\nCode id: "
+                                        + attachmentCodeID + "\n Unique Error ID: " + attachmentUniqueErrorCode
                                         + "\n In case there are violations they'll be present bellow ");
+                                sb.append(BundleUtil.getString(BUNDLE_CGDINTEGRATION,
+                                        "label.form43.sendForm43.problemsWhileUploadingAttachment", studentNumber.toString(),
+                                        attachmentValueMessage, attachmentCodeID.toString(), attachmentUniqueErrorCode.toString())
+                                        + "\n");
                                 for (ValidationResult validation : uploadFormAttachment.getViolations().getValue()
                                         .getValidationResult()) {
-                                    logger.error("Validation error : " + validation.getErrorMessage().getValue() + " [member: "
-                                            + validation.getMemberNames().getValue().getString().toString() + "]");
+                                    String attachmentValidationErrorMessage = validation.getErrorMessage().getValue();
+                                    String attachmentValidationMembers =
+                                            validation.getMemberNames().getValue().getString().toString();
+                                    logger.error("Validation error : " + attachmentValidationErrorMessage + " [member: "
+                                            + attachmentValidationMembers + "]");
+                                    sb.append(
+                                            BundleUtil.getString(BUNDLE_CGDINTEGRATION, "label.form43.sendForm43.validationError",
+                                                    attachmentValidationErrorMessage, attachmentValidationMembers) + "\n");
                                 }
                             } else {
-                                logger.info("Successful upload of form attachment for student with number:"
-                                        + registration.getStudent().getNumber());
+                                logger.info("Successful upload of form attachment for student with number:" + studentNumber);
+                                sb.append(BundleUtil.getString(BUNDLE_CGDINTEGRATION,
+                                        "label.form43.sendForm43.successfulUploadAttachment", studentNumber.toString()) + "\n");
                             }
                         }
                     }
                 }
+                CgdCommunicationLog log =
```

Comment:
@miguel-morais you are creating a domain object this should be ran inside a write TX